### DISCUSSION TITLE
Is it possible to tie a ProgressDialog to WriteAllBytes?

### DISCUSSION THREAD

**Discussion by thepwrtank18** [ookii-dialogs/ookii-dialogs-winforms#28] [Q&A] [answered]
While the program is doing WriteAllBytes to a file, the ProgressDialog will show it's progress.

**Comment by augustoproiete** [comment, marked as answer]
If by `WriteAllBytes` you mean [`File.WriteAllBytes`](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.writeallbytes?view=net-5.0), then no, it's not possible... This API is a one-shot API without any callbacks that would allow you to collect and report progress.

In order to report progress you have to take control of the writes by writing blocks of bytes to the stream yourself.
